routers/web/repo/editor.go

```diff
@@ -280,6 +297,10 @@ func editFilePost(ctx *context.Context, form forms.EditRepoFileForm, isNewFile b
 	operation := "update"
 	if isNewFile {
 		operation = "create"
+	} else if !form.Content.Has() {
+		// The form content only has data if file is representable as text, is not too large and not in lfs. If it doesn't
+		// have data, the only possible operation is a rename
+		operation = "rename"
```

Comment:
Maybe we also need to check whether the file name changed?

Reply:
7133834ffe1a19d81e9764db37fa344768e6cc95